buildtest/tools/menu.py

```diff
@@ -160,157 +145,6 @@ def get_menu(self):
 
         parser_get.set_defaults(func=func_get_subcmd)
 
-    def module_menu(self):
```

Comment:
Ok if we remove all these from argparse. Where will this be exposed. The module part is important but if it's not part of buildtest then will we have a separate program or will module API be taken out of buildtest into a separate git project.
The module load test is a key part of buildtest. Initially I wanted a framework to automate software stack testing

Reply:
The modules I think are best entirely controlled via the build test configuration files. If I need a particular module, I add it to the listing there. The state is determined at runtime, so it's not going to be dated or rely on some update to a cache. But I'm not required to use modules if I don't want to. The direction I think makes sense is to move toward a module being a kind of environment that can be used, but not so hard coded around the application, and with equal representation to other kinds of environments that the user might want. In the same way we will add different kinds of test config types, there can also be a family of environment/etc. types that can be specified for use to load for any particular build.

Reply:
>  Initially I wanted a framework to automate software stack testing
You would still have this! But to adjust your statement, what you were creating is _a framework to automate only stack testing if the site uses modules_. To be more true to the statement, it should be possible to support any flavor of test environment. Just like spack, or snakemake, in that you have a base class of "a test configuration case" and "an environment that can be used" I think we should aspire for that for buidltest. Then it will be the most useful / powerful.